src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java

```diff
@@ -120,4 +119,58 @@ private String generateIguassuToken(String userId) {
 
         return Base64.getEncoder().encodeToString(sessionToken.getBytes());
     }
+
+    public String refreshToken(String accessToken) throws Exception {
+        OAuthToken oAuthToken = this.iguassuFacade.getTokenByAccessToken(accessToken);
+        List<OAuthToken> tokens = this.iguassuFacade.getAllTokenByUserName(oAuthToken.getUserId());
+        if(Objects.nonNull(oAuthToken) && tokens.contains(oAuthToken)){
+            for(OAuthToken o : tokens){
+                if(o.getVersion() > oAuthToken.getVersion()){
+                    return o.getAccessToken();
+                }
+            }
+            OAuthToken refreshedToken = refreshToken(oAuthToken);
+            final String iguassuToken = this.generateIguassuToken(refreshedToken.getUserId());
+            this.storeOAuthToken(refreshedToken, iguassuToken);
+
+            final long oldVersions = oAuthToken.getVersion() - 1;
+            if(oldVersions >= 0){
+                this.iguassuFacade.removeOAuthTokens(oAuthToken.getUserId(), oldVersions);
+            }
+            return refreshedToken.getAccessToken();
+        } else {
+            throw new NotFoundAccessToken("The accessToken[" + accessToken +"] was not found");
```

Comment:
Treat this as an unauthorized operation.

Reply:
Okay. Done.